packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -267,9 +269,15 @@ Future<void> buildGradleApp({
 
   final BuildInfo buildInfo = androidBuildInfo.buildInfo;
   final String assembleTask = isBuildingBundle
-    ? getBundleTaskFor(buildInfo)
+    ? (isSetup ? 'buildRelease': getBundleTaskFor(buildInfo))
     : getAssembleTaskFor(buildInfo);
 
+  List<String> extraGenSnapshotOptions = buildInfo.extraGenSnapshotOptions ?? <String>[];
+  if (androidBuildInfo.splitAot) {
```

Comment:
you could also just check if splitAot is defined in the targets/ and conditionally add to the gen snapshot options